lib/config/config-ops.js

```diff
@@ -268,5 +272,84 @@ module.exports = {
+     * @param {Array<number|string>} vector array of config file paths or relative override indices
+     * @returns {Object} config object
+     */
+    getConfigFromVector(vector) {
+
+        const subvector = Array.from(vector);
+        let config,
+            subConfigs,
+            nearestCacheIndex = vector.length - 1;
+
+        // Check the merged vector config cache to try to find a merged config for the current config or a parent
+        for (; nearestCacheIndex >= 0; nearestCacheIndex--) {
+            config = configCache.getMergedVectorConfig(subvector);
+            if (config) {
+                break;
+            }
+            subvector.pop();
+        }
+
+        if (config) {
+            if (nearestCacheIndex === vector.length - 1) {
+                return config;
+            }
+            debug("Using config from partial cache");
+
+            // Get parent config if configKey is an override index
+            if (typeof vector[nearestCacheIndex + 1] === "number") {
+
+                // If the first non-cached vector is an override index, subConfigs needs to be set
+                let parentConfigKey = vector[nearestCacheIndex];
+
+                for (let i = nearestCacheIndex - 1; i >= 0 && typeof parentConfigKey !== "string"; i--) {
+                    parentConfigKey = vector[i];
+                }
+                subConfigs = configCache.getConfig(parentConfigKey).overrides;
+            }
+        } else {
+            config = {};
+        }
+
+        // Start from index of nearest cached config
+        for (let i = nearestCacheIndex + 1; i < vector.length; i++) {
+            const configKey = vector[i];
+            let shallowConfig;
+
+            if (typeof configKey === "string") {
+                shallowConfig = configCache.getConfig(configKey);
+                subConfigs = shallowConfig.overrides;
+            } else {
+                shallowConfig = subConfigs[configKey];
+            }
+            config = this.merge(config, shallowConfig);
+            if (config.filePath) {
+                delete config.filePath;
+                delete config.baseDirectory;
+            } else if (config.files) {
+                delete config.files;
+            }
+            configCache.setMergedVectorConfig(vector, config);
+        }
+
+        return config;
+    },
+
+    /**
+     * Checks that the specified file path matches all of the supplied glob patterns.
+     * @param {string} filePath The file path to test patterns against
+     * @param {string|string[]} patterns One or more glob patterns, all of which will be matched against the file path
+     * @returns {boolean} True if all the supplied patterns match the file path, false otherwise
+     */
+    pathMatchesGlobs(filePath, patterns) {
+        patterns = [].concat(patterns);
+        return patterns.every(pattern => minimatch(filePath, pattern, { matchBase: true }));
```

Comment:
We are not protecting against "../" patterns anywhere. Those will basically never work (or at least will not work in most cases). They are also going to be really hard to wrap your head around, so I think we should filter out and ignore them.